Classes/Controller/PersonController.php

```diff
@@ -208,92 +241,35 @@ public function newAction(\Personmanager\PersonManager\Domain\Model\Person $newP
     /**
      * action create
      *
-     * @param \Personmanager\PersonManager\Domain\Model\Person $newPerson
+     * @Extbase\Validate(param="newPerson", validator="Personmanager\PersonManager\Domain\Validator\HoneyPotValidator")
+     * @Extbase\Validate(param="newPerson", validator="Personmanager\PersonManager\Domain\Validator\TermsValidator")
+     * @Extbase\Validate(param="newPerson", validator="Personmanager\PersonManager\Domain\Validator\Person\NameValidator")
+     * @Extbase\Validate(param="newPerson", validator="Personmanager\PersonManager\Domain\Validator\Person\EmailValidator")
+     * @Extbase\Validate(param="newPerson", validator="Personmanager\PersonManager\Domain\Validator\Person\EmailAgainValidator")
+     * @param Person $newPerson
      * @return void
      */
-    public function createAction(\Personmanager\PersonManager\Domain\Model\Person $newPerson)
+    public function createAction(Person $newPerson)
     {
-        //TODO create Validators for Error Handling
-        $failed = 0;
-        $honey = $this->request->getArguments()["tx_personmanager_personmanagerfront"]["honeypot"];
-        $anr = $this->request->getArguments()["anr"];
-        $terms = $this->request->getArguments()["terms"];
-
-        $termVar = $GLOBALS['TSFE']->tmpl->setup["plugin."]["tx_personmanager."]["variables."]["terms"];
-
-        $newPerson->setSalutation($anr);
-        $error = "";
-        $newPerson->setEmail(trim($newPerson->getEmail()));
-
-        if ($honey != "" && $honey != NULL) {
-            $langhelp = LocalizationUtility::translate('error.spam', $this->extKey);
-            $error .= "<p>$langhelp</p>";
-            $failed = 1;
-        }
-        if ($termVar) {
-            if ($terms != "1" || $terms != 1) {
-                $langhelp = LocalizationUtility::translate('error.terms', $this->extKey);
-                $error .= "<p>$langhelp</p>";
-                $failed = 1;
-            }
-        }
-        if ($newPerson->getLastname() == "" || $newPerson->getLastname() == NULL || $newPerson->getFirstname() == "" || $newPerson->getFirstname() == NULL) {
-            $langhelp = LocalizationUtility::translate('error.name', $this->extKey);
-            $error .= "<p>$langhelp</p>";
-            $failed = 1;
-        }
-        if (!filter_var(idn_to_ascii($newPerson->getEmail(),0,INTL_IDNA_VARIANT_UTS46), FILTER_VALIDATE_EMAIL)) {
-            $langhelp = LocalizationUtility::translate('error.email', $this->extKey);
-            $error .= "<p>$langhelp</p>";
-            $failed = 1;
-        }
+        $hash = $newPerson->getEmail() . time();
+        $newPerson->setToken(md5($hash));
+
         $oldMail = $this->personRepository->findOneByEmail($newPerson->getEmail());
```

Comment:
DB Request already made in EmailAgainValidator
maybe remove oldMail in Validator?